lib/commands/exam.js

```diff
@@ -161,12 +161,17 @@ module.exports = TestCommand.extend({
     const events = config.custom_browser_socket_events || {};
     const testExecutionFileName = `test-execution-${Date.now()}.json`;
     const getBrowserId = function(testPage) {
```

Comment:
It feels like setting `command.executionOnly = true;`should be done elsewhere outside of `getBrowserId()`,  some sorta setup step. 

Do you think something like this would make sense?
```
    const getBrowserId = function() {
      // if noExecutionFile is true, the browserId should not matter, returning 0 
      if (commands.noExecutionFile) return 0;

      const browserId = /browser=\s*([0-9]*)/.exec(this.launcher.settings.test_page);

      if (Array.isArray(browserId) !== null) {
        return browserId[1];
      } else {
        throw new Error(`Browser Id couldn't be found from the test_page: ${this.launcher.settings.test_page}`);
      }
    }

...
      events['tests-start'] = function() {
        // 'tests-start' is emitted everytime a test starts.
        // To prevent from registering `processExit` and `processError` multiple times set a flag to check if callbacks are registered.
        if (!registerProcessOn) {
          if (this.process && this.launcher) {
            this.process.on('processExit', browserFailureHandler.bind(this));
            this.process.on('processError', browserFailureHandler.bind(this));
          } else {
            // if this.process or this.launcher is undefined, it means `no-launch` was passed
            // when running with `--no-launch`, a test execution file is not created
            // because Testem does not create browser instance, which is needed to correctly
            // book keep the test execution
            commands.noExecutionFile = true;
          }
          registerProcessOn = true;
        }
```